stdlib/benchmarks/builtin/bench_hash.mojo

```diff
@@ -0,0 +1,872 @@
+волосы, последние, брата, месяца
+"""
+
+
+fn gen_word_pairs[words: String = words_en]() -> List[String]:
+    var result = List[String]()
+    try:
+        var list = words.split(",")
+        for w in list:
+            var w1 = w[].strip()
+            for w in list:
+                var w2 = w[].strip()
+                result.append(w1 + " " + w2)
+    except:
+        pass
+    return result
+
+
+# ===----------------------------------------------------------------------===#
+# AHash based hash function
+# ===----------------------------------------------------------------------===#
+
+from bit import byte_swap
+from bit import rotate_bits_left
+
+alias U256 = SIMD[DType.uint64, 4]
+alias U128 = SIMD[DType.uint64, 2]
+alias MULTIPLE = 6364136223846793005
+alias ROT = 23
+
+
+@always_inline
+fn _folded_multiply(lhs: UInt64, rhs: UInt64) -> UInt64:
+    """A fast function to emulate a folded multiply of two 64 bit uints.
+    Used becuase we don't have UInt128 type.
+
+    Args:
+        lhs: 64 bit uint.
+        rhs: 64 bit uint.
+
+    Returns:
+        A value which is similar in its bitpattern to result of a folded multply.
+    """
+    var b1 = lhs * byte_swap(rhs)
+    var b2 = byte_swap(lhs) * (~rhs)
+    return b1 ^ byte_swap(b2)
+
+
+@always_inline
+fn read_small(data: UnsafePointer[UInt8], length: Int) -> U128:
+    """Produce a SIMD[DType.uint64, 2] value from data which is smaller or equal 8 bytes.
+
+    Args:
+        data: Pointer to the byte array.
+        length: The byte array length.
+
+    Returns:
+        Returns a SIMD[DType.uint64, 2] value.
+    """
+    if length >= 2:
+        if length >= 4:
+            # len 4-8
+            var a = data.bitcast[DType.uint32]().load().cast[DType.uint64]()
+            var b = data.offset(length - 4).bitcast[DType.uint32]().load().cast[
+                DType.uint64
+            ]()
+            return U128(a, b)
+        else:
+            # len 2-3
+            var a = data.bitcast[DType.uint16]().load().cast[DType.uint64]()
+            var b = data.offset(length - 1).load().cast[DType.uint64]()
+            return U128(a, b)
+    else:
+        # len 0-1
+        if length > 0:
+            var a = data.load().cast[DType.uint64]()
+            return U128(a, a)
+        else:
+            return U128(0, 0)
+
+
+struct AHasher:
+    var buffer: UInt64
+    var pad: UInt64
+    var extra_keys: U128
+
+    fn __init__(inout self, key: U256):
+        """Initialise the hasher with a key.
+
+        Args:
+            key: Modifier for the computaiton of final hash value.
+        """
+        var pi_key = key ^ U256(
+            0x243F_6A88_85A3_08D3,
+            0x1319_8A2E_0370_7344,
+            0xA409_3822_299F_31D0,
+            0x082E_FA98_EC4E_6C89,
+        )
+        self.buffer = pi_key[0]
+        self.pad = pi_key[1]
+        self.extra_keys = U128(pi_key[2], pi_key[3])
+
+    @always_inline
+    fn large_update(inout self, new_data: U128):
```

Comment:
@mzaks, probably this is intended to be internal, so not part of the API, so, why not starting with `_`?

```suggestion
    fn _large_update(inout self, new_data: U128):
```